tools/registry.py

```diff
@@ -71,6 +73,53 @@ def json_dump(file, data):
     json.dump(data, f, indent=4, sort_keys=True)
     f.write("\n")
 
+# Translated from:
+# https://github.com/bazelbuild/bazel/blob/79a53def2ebbd9358450f739ea37bf70662e8614/src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Version.java#L58
+@functools.total_ordering
+class Version:
+
+  @functools.total_ordering
+  class Identifier:
+    def __init__(self, s):
+      self.val = int(s) if s.isnumeric() else s
```

Comment:
nit: `s` could be empty here if the version looks like `1.2..3`, which is illegal

Reply:
But `"".isnumeric()` will be false, so it's fine, I mean the script won't crash, right?

Reply:
I see, done!